doc/md/guides/mysql/ds-103.md

```diff
@@ -0,0 +1,136 @@
+---
+id: mysql-ds-103
+title: Preventing accidental column deletion on MySQL
+slug: /guides/mysql/checks/DS103
+tags: [linting, ci, destructive-change, mysql, drop-column]
+---
+
+## Introduction
+
+This document describes in detail the detection of dropping non-virtual
+columns for MySQL databases by Atlas's lint capabilities.
+
+MySQL allows users to modify existing table structure using the 
+[`ALTER TABLE`](https://dev.mysql.com/doc/refman/8.0/en/alter-table.html)
+statement. Using this statement, users can drop columns that are
+no longer needed. For example:
+
+```text
+ALTER TABLE t DROP COLUMN c;
+```
+
+While this is sometimes desired, deleting a column from an active database
+can have various forms negative impact on application which we describe below.
+
+To prevent damage to production databases, many teams employ a policy that
+prohibits dropping columns (or database resources altogether).  Atlas's 
+[migrate lint](/versioned/lint) can help teams automatically detect migrations
+that drop columns during continuous integration. For example:
+
+```text
+atlas migrate lint --dev-url mysql://root:pass@localhost:3306/dev --latest 1
+```
+
+May detect:
+
+```text
+20221023123017.sql: destructive changes detected:
+
+	L1: Dropping non-virtual column "c2"
+```
+
+## Impact of dropping columns
+
+### Data loss
+
+Destructive changes are changes to a database schema that result in loss of data.
+For instance, consider a statement such as:
+
+```sql
+ALTER TABLE `users` DROP COLUMN `email_address`;
+```
+
+This statement is considered destructive because whatever data is stored 
+in the email_address column will be deleted from disk, with no way to recover it.
+If this happens accidentally during a schema change, the business
+impact of this change may be dire. 
+
+Contrary to many other types of changes, dropping a column is not a reversible
+operation. In other words, the "reverse" statement of the above statement is
+not a symmetric operation. Consider this statement:
+
+```sql
+ALTER TABLE `users` ADD COLUMN `email_address` varchar(255);
+```
+
+Running this query may "roll-back" the column drop operation, but any
+data which was stored in this column will be lost.
+
+### Backwards Incompatibility
+
+In the past, it was acceptable in many situations to stop an application
+for short maintenance windows during which database administrators could
+apply changes to the database schema. After the changes were applied a new
+version of the application which expects the new schema would be started. 
```

Comment:
Rewrite sentence? I think you're missing a piece :)

Reply:
I think the sentence is correct.. how would you say it?